luarules/gadgets/unit_air_plants.lua

```diff
@@ -74,6 +87,59 @@ local airCmd = {
 	params = { '1', 'LandAt 0', 'LandAt 30', 'LandAt 50', 'LandAt 80' }
 }
 
+-- Fallback method to make sure units don't land atop their origin plant.
+local function moveFromPlant(unitID, factoryRadius)
+	local ux, uy, uz = Spring.GetUnitPosition(unitID)
+	local theta = math.random() * 2 * math.pi
+	local radius = 2.5 * factoryRadius
+	local tx, tz = ux + radius * math.sin(theta), uz + radius * math.cos(theta)
+	local ty = Spring.GetGroundHeight(tx, tz)
+	GiveOrderToUnit(unitID, CMD.INSERT, { 0, CMD.MOVE, 0, tx, ty, tz }, 0)
+end
+
+-- Prevent air units from relaxing their target height when leaving the plant.
+local function moveToCommand(unitID, command, factoryRadius)
+	local tx, ty, tz
+	if command then
+		if #command.params == 1 then
+			if command.params[1] > Game.maxUnits then
+				tx, ty, tz = Spring.GetFeaturePosition(command.params[1])
+			else
+				tx, ty, tz = Spring.GetUnitPosition(command.params[1])
+			end
+		elseif #command.params >= 3 then
+			tx, ty, tz = command.params[1], command.params[2], command.params[3]
+		end
+	end
+
+	if not tx then
+		moveFromPlant(unitID, factoryRadius)
+		return
+	end
+
+	-- Within a short distance, air units do not dip when leaving the plant.
+	local distanceMin = Spring.GetUnitMoveTypeData(unitID).turnRadius or 0
+	if distanceMin == 0 or distanceMin == bomberAttackTurnRadius then
+		distanceMin = unitMoveRadius[Spring.GetUnitDefID(unitID)]
+	end
+	distanceMin = math.max(distanceMin, factoryRadius)
+
+	local ux, uy, uz = Spring.GetUnitPosition(unitID)
+	local dx = tx - ux
+	local dy = ty - uy
+	local dz = tz - uz
+	local distanceSq = dx * dx + dy * dy + dz * dz
+
+	if distanceSq > distanceMin * distanceMin then
+		-- Nudge the unit toward its command without sending it too far.
+		local scale = math.sqrt(distanceMin * distanceMin / distanceSq)
```

Comment:
consider avoiding the squaring/rooting
```suggestion
	local distance = math.diag(dx, dy, dz)

	if distance > distanceMin then
		-- Nudge the unit toward its command without sending it too far.
		local scale = distanceMin / distance
```